edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceScopeApiClient.cs

```diff
@@ -133,6 +154,60 @@ async Task<ScopeResult> GetIdentitiesInScope(Uri uri)
             }
         }
 
+        async Task<ScopeResult> GetIdentitiesInTargetScopeWithRetry(Uri uri, Option<string> continuationToken)
+        {
+            try
+            {
+                return await ExecuteWithRetry(
+                    () => this.GetIdentitiesInTargetScope(uri, continuationToken),
+                    Events.RetryingGetIdentities,
+                    this.retryStrategy);
+            }
+            catch (Exception e)
+            {
+                Events.ErrorInScopeResult(e);
+                throw;
+            }
+        }
+
+        async Task<ScopeResult> GetIdentitiesInTargetScope(Uri uri, Option<string> continuationLink)
+        {
+            HttpClient client = this.proxy
+                .Map(p => new HttpClient(new HttpClientHandler { Proxy = p }, disposeHandler: true))
+                .GetOrElse(() => new HttpClient());
+            using (var msg = new HttpRequestMessage(HttpMethod.Post, uri))
+            {
+                if (!this.AuthChainProvider.TryGetAuthChain(this.deviceId, out string authChain))
+                {
+                    throw new ArgumentException($"No valid authentication chain for {this.deviceId}");
+                }
+
+                var payload = new NestedScopeRequest()
+                {
+                    AuthChain = authChain,
+                    PageSize = this.batchSize,
+                    ContinuationLink = continuationLink.GetOrElse(() => string.Empty)
+                };
+
+                string token = await this.edgeHubTokenProvider.GetTokenAsync(Option.None<TimeSpan>());
+                msg.Headers.Add(HttpRequestHeader.Authorization.ToString(), token);
+                msg.Content = new StringContent(JsonConvert.SerializeObject(payload).ToString(), Encoding.Default, "application/json");
+
+                HttpResponseMessage response = await client.SendAsync(msg);
+                string content = await response.Content.ReadAsStringAsync();
+                if (response.IsSuccessStatusCode)
+                {
+                    var scopeResult = JsonConvert.DeserializeObject<ScopeResult>(content);
+                    Events.GotValidResult();
+                    return scopeResult;
+                }
+                else
+                {
+                    throw new DeviceScopeApiException("Error getting device scope result from IoTHub", response.StatusCode, content);
```

Comment:
should we change the word "IoTHub", as it can be nested parent edge device? or even better, if we have different message for iot hub and parent edge device (including parent edge device id in the message).
```suggestion
                    throw new DeviceScopeApiException("Error getting device scope result from upstream", response.StatusCode, content);
```